api/src/helpers/transferCredits.ts

```diff
@@ -362,44 +349,27 @@ export const organizeLegacyTransfers = async (rows: TransferData[]) => {
   // Validate all the relevant courses from the API beforehand
   const coursesToValidate = new Set<string>();
   for (const transfer of transfers) {
-    const transferName = transfer.courseName.trim();
-    if (transferName.startsWith('AP ')) {
+    if (transfer.courseName.startsWith('AP ')) {
       continue;
     }
-    coursesToValidate.add(removeAllSpaces(transferName));
+    coursesToValidate.add(removeAllSpaces(transfer.courseName));
   }
-  // console.log('Validating courses from the API (batch)');
   const validCourses = await validateCoursesAPI(coursesToValidate);
```

Comment:
similar idea here
`const validCourses = getValidCourseNames(transfers)`

would call a helper that builds the courses to validate set and loops through all transfers